Update content-style-guide.md (#2400)
fishtown-analytics/docs.getdbt.com: contributing/content-style-guide.md

* [Images](#Images)
* [Talk to us](#Talk-to-us)

## Folder Structure and TOC

The folder structure for the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository is organized into several high-level categories under the main `website` folder: blog, cypress, docs, functions, plugins, snippets, src, static.

The table of contents (TOC) is managed in the `sidebar.js` file. You only need to edit the `sidebar.js` file when you are adding a new page or deleting an existing page. The `sidebar.js` file is the one that causes most of the merge conflicts because many technical writers are working on content daily. You will need to accept the changes from other contributors if you are committing a PR.

Don't worry if you're not sure where in the TOC a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.

The right-hand TOC is created automatically when you add headings to a page.

## Filenaming

> If you are adding a new file in the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository, it must be named following our naming conventions. MORE TO ADD

Reviewer: Should we define our naming conventions before we add this line (remove the "MORE TO ADD")?

Author: yes def! we can try to decide on this async but that's another one of my questions in the description up top!  maybe i can start a thread in slack to get it started?

Author: Ideas for content once polling is finished:

When you create or rename a file or directory and it has more than one word in its name, use dashes (-) instead of spaces or underscores. 

**Option 1:**
The file name should always start with a verb and the feature type, like ide, scheduler, or python. Depending on the content type, it typically also includes a secondary descriptor. Verbs are used when you are creating a procedural document. 

Due to the fact that we author content using Markdown, you must add the `.md` file extension to the file name. For example, `ide-features.md`

If you are adding a new file to an existing feature category, follow the current naming convention.

| File type | Example |
| :-------: | :-------: |
| Conceptual file | `set-up-dbt-cloud.md` |
| Procedural file | `set-up-cloud-environment.md` |
| Reference file | `exclude-flag.md` |

**Option 2:**
The file name should always start with the feature type, like ide, scheduler, or python. Depending on the content type, it typically also includes a secondary descriptor or a verb. Verbs are used when you are creating a procedural document. 

Due to the fact that we author content using Markdown, you must add the `.md` file extension to the file name. For example, `ide-features.md`

If you are adding a new file to an existing feature category, follow the current naming convention.

| File type | Example |
| :-------: | :-------: |
| Conceptual file | `dbt-cloud-set-up.md` |
| Procedural file | `cloud-environment-set-up.md` |
| Reference file | `exclude-flag.md` |

> ## Using Markdown

Reviewer: This whole section is awesome and very useful.  If/when we rearrange the order of these sections, I think this should be at the top.

Don't worry if you're not sure where in the TOC a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.

The right-hand TOC is created automatically when you add headings to a page.

## Filenaming

If you're adding a new file in the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository, review the following file name guidance and examples: 

- Keep the file name as short as possible since it becomes part of the URL and leave out any unnecessary words. Long URLs are hard to read and prone to errors.
- Think about where you want to locate the file in the leftbar. This will help catch and avoid redundancy. Remain consistent with the surrounding leftbar items.
- Be as descriptive as possible so readers have an idea as to what they’re about to read.

| ✅ User | ❌ Avoid|
|:-------:|:-------:|

> |`/docs/deploy/architecture` | `/docs/deploy/about-architecture`|

Reviewer 2: I would be OK with `about-architecture.md` maybe a better thing to avoid is something like `how-we-think-about-architecture-at-dbt.md` 😆

Author: thanks @runleonarun , oh gotcha!  i used one of @nghi-ly 's suggestions on this one but maybe it makes sense to include that example you shared too 🙈

> |`/docs/deploy/architecture` | `/docs/deploy/about-architecture`|

Author: ```suggestion
|`/docs/deploy/architecture` | `/docs/deploy/how-we-think-about-architecture`|
```

| Code or command in a sentence               | ``code``                                              |
| Link - external site                        | `[Title](https://www.example.com)`                    |
| Link - topic in same folder                 | `[Title](/folder/file-name) without file extension`            |
| Link - topic in different folder            | `[Title](/folder/file-name) without file extension` |
| Link - section in topic in same folder      | `[Title](/folder/file-name#section-name)`                     |
| Link - section in topic in different folder | `[Title](/folder/file-name#section-name)`           |
| Image                                       | `<Lightbox src="/img/docs/<image-name>.jpg" title="Concise description of image"/>`      |

## Callouts

Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!

> Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts. Below are the some of the typical callout formats we use:

Reviewer 2: This looks great! We might also want to say that if you add text after the first code, like this: `:::note This is a note title` it shows up as a title for the note.

## Using Markdown

docs.getdbt.com uses its own CSS, and Docusaurus supports its own specific Markdown syntax. The following table provides an overview of the supported syntax elements.

| Element                                     | Syntax                                                |
|---------------------------------------------|-------------------------------------------------------|
| Headings                                    | `# H1`, `## H2`, `### H3`                             |
| Bold                                        | `**bold text**`                                       |
| Italic                                      | `_italicized text_`                                    |
| Ordered List                                | `1.` First item (use `1.` for each item)              |
| Unordered List                              | `-` or `*` (for each item)                            |
| Code or command in a sentence               | ``code``                                              |
| Link - external site                        | `[Title](https://www.example.com)`                    |

> | Link - topic in same folder                 | `[Title](/folder/file-name) without file extension`            |

Reviewer 3: i don't believe we want to use this ^ link format. i think we want the "/docs/<path-to-file>" format

Author: ah ok gotcha, i kept it generic as '/folder/file-name' as it's different for guides and other folders (unless that still starts with /docs/guides...?)

I've changed it to /docs though but will bring it up at our session

> | Link - topic in different folder            | `[Title](/folder/file-name) without file extension` |

Reviewer 3: i think the same thing with this ^

> | Link - section in topic in same folder      | `[Title](/folder/file-name#section-name)`                     |

Reviewer 3: same thing with this ^

> | Link - section in topic in different folder | `[Title](/folder/file-name#section-name)`           |

Reviewer 3: same thing with this ^

| Image                                       | `<Lightbox src="/img/docs/<image-name>.jpg" title="Concise description of image"/>`      |

> 

Reviewer 3: sugg:

- linking to https://www.markdownguide.org/basic-syntax/ for basic Markdown syntax instead of listing them in the table

reason: 

- for conciseness and easier docs maintenance. not too lengthy of a style guide and changes to basic Markdown are maintained elsewhere 
- to have content be focused on what's different or custom with ours ("our stuff"). include things like: internal links should be in the format of "/docs/<path-to-file>" and Lightbox (which is custom, i think) and so on.

## Callouts

> Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!

Reviewer 2: Moving the note about the title from below to here. 

```suggestion
Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!

If you add text after the first code, like this `:::note This is a note title`, it shows up as a title for the note.
```

> Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts. Below are the some of the typical callout formats we use:

Reviewer 3: ```suggestion
Below are the some of the typical callout formats we use:
```

Reviewer 3: sugg: second sentence in previous paragraph mentions this too

Reviewer 2: I like Ly's sugg and we can even make it shorter because I think these are the only formats possible. 
```suggestion
Callout formats include:
```

Author: good shout! the paragraph previously existed and i only added the last sentence. however, i love that we can shorten it!

| Types of callouts | Callout formats |
| ---- | ------ |
| Note callouts are used for notices| ```:::note``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
| Info callouts are used to highlight info |```:::info``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
| Tip callouts are used for tips |```:::tip``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
| Caution callouts are used for warnings/considerations |```:::caution``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |

> 

Reviewer 3: <3

* [Images](#Images)
* [Talk to us](#Talk-to-us)

## Folder Structure and TOC

> The folder structure for the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository is organized into several high-level categories under the main `website` folder: blog, cypress, docs, functions, plugins, snippets, src, static.

Reviewer 3: ```suggestion
The folder structure for the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository is organized into several high-level categories under the main `website` folder such as `blog`, `cypress`, `docs`, `functions`, `plugins`, `snippets`, `src`, `static`, and so on.
```

> The table of contents (TOC) is managed in the `sidebar.js` file. You only need to edit the `sidebar.js` file when you are adding a new page or deleting an existing page. The `sidebar.js` file is the one that causes most of the merge conflicts because many technical writers are working on content daily. You will need to accept the changes from other contributors if you are committing a PR.

Reviewer 3: ```suggestion
The left sidebar (leftbar) is managed in the `sidebar.js` file. You only need to edit the `sidebar.js` file when you are adding a new page or deleting an existing page. Editing the `sidebar.js` file can cause merge conflicts as it's updated often because content is being worked on daily. You will need to accept the changes from other contributors if you are committing a PR.
```

> Don't worry if you're not sure where in the TOC a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.

Reviewer 3: ```suggestion
Don't worry if you're not sure where in the leftbar a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.
```

> The right-hand TOC is created automatically when you add headings to a page.

Reviewer 3: ```suggestion
The right sidebar (rightbar), also known as the page TOC, is created automatically for all the H2 and H3 headings on a page. For details, see [Headings and Table of contents](https://docusaurus.io/docs/markdown-features/toc) in Docusaurus. 
```

| Image                                       | `<Lightbox src="/img/docs/<image-name>.jpg" title="Concise description of image"/>`      |

## Callouts

Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!

Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts. Below are the some of the typical callout formats we use:

| Types of callouts | Callout formats |
| ---- | ------ |
| Note callouts are used for notices| ```:::note``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
| Info callouts are used to highlight info |```:::info``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
| Tip callouts are used for tips |```:::tip``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
| Caution callouts are used for warnings/considerations |```:::caution``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |

> Note: If you add text after the first code, like this `:::note This is a note title`, it will show up as a title for the note.

Reviewer 2: Just moving this to the top so it doesn't get lost.
```suggestion
```

> | `-file` | Print output to file instead of stdout. | Name of the file. |

## Word choice & terminology

> Use active voice instead of passive. Active voice is clearer and more direct, making it easier to translate. 

Reviewer 2: GREAT addition! 🙇🏻‍♀️

* Use screenshots to highlight navigation, on-screen elements, and other noteworthy product visuals.
* Avoid using screenshots to demonstrate inputs and outputs. All code snippets and sample results should be in the documents as text fields.
* Add images are under the `static` -> `img` folder.
* Use concise filenames that are relevant to the content contained within. Enumerate them if they are part of a sequence.

> * Use PNG or JPEG format, which renders a better quality and lossless compression.

Reviewer: Should we remove PNG based on the feedback around dark mode support?

Author: i think so too! i posed that question up top in the description and IMO it makes sense as the win-win option but def wanted to double check!

> * Add *title=""* for all images to write a consice title of the image. For accessibility, it's important to use succinct text that is clear and complete. 

Reviewer: ```suggestion
* Add *title=""* for all images to write a concise title of the image. For accessibility, it's important to use succinct text that is clear and complete. 
```

* Use screenshots to highlight navigation, on-screen elements, and other noteworthy product visuals.
* Avoid using screenshots to demonstrate inputs and outputs. All code snippets and sample results should be in the documents as text fields.
* Add images are under the `static` -> `img` folder.
* Use concise filenames that are relevant to the content contained within. Enumerate them if they are part of a sequence.
* Use JPEG format, which renders a better quality and lossless compression. JPEG format has a white background and is accessible in light and dark mode. 

> * Add *title=""* for all images to write a concise title of the image. For accessibility, it's important to use succinct text that is clear and complete. 

Reviewer 2: Should we move [this content](https://www.notion.so/dbtlabs/Lightbox-Component-19d4496c073542868dee7bf33ff59b30) to the `contributing/` folder so we can link to it?

Author: good callout!
